gen/field_test.go

```diff
@@ -76,3 +78,54 @@ func TestFieldLabelConflict(t *testing.T) {
 		})
 	}
 }
+
+func TestCompileJSONTag(t *testing.T) {
+	tests := []struct {
+		desc             string
+		required         bool
+		options          []string
+		isOmitemptyAdded bool
+	}{
+		{
+			desc:             "for optional fields",
+			required:         false,
+			options:          make([]string, 0),
```

Comment:
`nil` is a valid empty slice. This line can be deleted.

https://github.com/uber-go/guide/blob/master/style.md#nil-is-a-valid-slice

```suggestion
```